pkg/liquidity-mining/contracts/admin/GaugeAdder.sol

```diff
@@ -91,95 +134,89 @@ contract GaugeAdder is IGaugeAdder, SingletonAuthentication, ReentrancyGuard {
      */
-    function addGaugeFactory(ILiquidityGaugeFactory factory, GaugeType gaugeType) external override authenticate {
-        // Casting is safe as n_gauge_types return value is >= 0.
-        require(uint256(gaugeType) < uint256(_gaugeController.n_gauge_types()), "Invalid gauge type");
+    function addGaugeFactory(ILiquidityGaugeFactory factory, string memory gaugeType) external override authenticate {
+        bytes32 gaugeTypeBytes = _validateAndCastGaugeType(gaugeType);
 
         // Sanity check that calling `isGaugeFromFactory` won't revert
         require(!factory.isGaugeFromFactory(address(0)), "Invalid factory implementation");
 
-        EnumerableSet.AddressSet storage gaugeFactories = _gaugeFactoriesByType[gaugeType];
+        EnumerableSet.AddressSet storage gaugeFactories = _gaugeFactoriesByType[gaugeTypeBytes];
         require(gaugeFactories.add(address(factory)), "Factory already added");
 
         emit GaugeFactoryAdded(gaugeType, factory);
     }
 
+    // TODO: optional admin functions.
+
+    function setGaugeTypeNumber(string memory gaugeType, int128 gaugeTypeNumber) external authenticate {
+        // TODO: implement.
+        // This would change the gauge type number for an existing gauge type.
+    }
+
+    function removeGaugeType(string memory gaugeType) external authenticate {
+        // TODO: implement.
+        // This would remove an existing gauge type.
+        // It would not clear the existing factories.
+    }
+
     // Internal functions
 
     /**
      * @dev Adds `gauge` to the GaugeController with type `gaugeType` and an initial weight of zero
      */
-    function _addGauge(address gauge, GaugeType gaugeType) private {
-        require(isGaugeFromValidFactory(gauge, gaugeType), "Invalid gauge");
+    function _addGauge(address gauge, bytes32 gaugeType) private {
+        require(_isGaugeFromValidFactory(gauge, gaugeType), "Invalid gauge");
+        int128 gaugeTypeNumber = _gaugeTypeNumber[gaugeType];
 
         // `_gaugeController` enforces that duplicate gauges may not be added so we do not need to check here.
         _authorizerAdaptorEntrypoint.performAction(
             address(_gaugeController),
-            abi.encodeWithSelector(IGaugeController.add_gauge.selector, gauge, gaugeType)
+            abi.encodeWithSelector(IGaugeController.add_gauge.selector, gauge, gaugeTypeNumber)
         );
     }
+
+    function _stringToBytes32(string memory str) internal pure returns (bytes32 bytesString) {
+        require(bytes(str).length <= 32, "Input string should be 32 characters long at the most");
+
+        assembly {
+            bytesString := mload(add(str, 32))
+        }
+    }
+
+    function _bytes32ToString(bytes32 _bytes32) internal pure returns (string memory) {
+        bytes memory bytesArray = new bytes(32);
+        for (uint256 i = 0; i < 32; ++i) {
+            bytesArray[i] = _bytes32[i];
+        }
+
+        return string(bytesArray);
+    }
```

Comment:
This is sad, but that's the way it is in 0.7.1.

In 0.8.x there are better conversions between the types.